pkg/services/sso/redhatsso_service.go

```diff
@@ -290,6 +291,22 @@ func (r *redhatssoService) CreateServiceAccountInternal(accessToken string, requ
 }
 
 func (r *redhatssoService) DeleteServiceAccountInternal(accessToken string, clientId string) *errors.ServiceError {
+	if strings.HasPrefix(clientId, "canary") || strings.HasPrefix(clientId, "kas-fleetshard-agent") {
+		first := 0
+		max := 100
+		for {
+			accounts, err := r.client.GetServiceAccounts(accessToken, first, max)
```

Comment:
What's the performance impact of doing this call each time we delete a kafka or OSD cluster?